### ISSUE TITLE
Can't create a factory from a workspace with SVN project

### ISSUE THREAD

**Issue by ddementieva** [codenvy/codenvy#687]
Can't create a factory from a workspace with SVN project

**Reproduction Steps:**

1. Open any workspace in the IDE and import any svn repo there. 
2. Go to the Factories and try to create Factory using workspace with svn repo in it. 

**Expected behavior:**

Factory is created.

**Observed behavior:**

The following error appear: "You have provided an invalid parameter source.project.parameters.password for this version of Factory parameters 4.0. For more information, please visit http://docs.codenvy.com/user/project-lifecycle/#configuration-reference"

**Che version:**    4.7.0

**Additional information:**

* Problem started happening recently, didn't happen in an older version of Che: Yes
* Problem can be reliably reproduced, doesn't happen randomly: Yes

**Comment by ddementieva** [comment]
curl 'https://beta.codenvy.com/api/factory' -H 'Origin: https://beta.codenvy.com' -H 'Accept-Encoding: gzip, deflate, br' -H 'Accept-Language: en-US,en;q=0.8' -H 'User-Agent: Mozilla/5.0 (X11; Linux x86_64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/53.0.2785.92 Safari/537.36' -H 'Content-Type: multipart/form-data; boundary=----WebKitFormBoundaryYpGppVWGMw0UYAj3' -H 'Accept: application/json, text/plain, */*' -H 'Referer: https://beta.codenvy.com/dashboard/' -H 'Cookie: JSESSIONID=7EB7F4150E06F9D2B955B191EE9EDC50; __utma=28572719.161833912.1473077541.1473077541.1473077541.1; __utmc=28572719; __utmz=28572719.1473077541.1.1.utmcsr=(direct)|utmccn=(direct)|utmcmd=(none); session-access-key=yynryfqfjGaOebWueLeeTeHzHnjCS11u8e4mfPH994Sn0mvGzvP590fCHCuqbCzTv9uvbfnHKTaiLKLCavarCjTHWz9Kv1zvGDzfXbrPyG1CzOuXSnHrzm9mfL0H0X0SXauue4SyDnH9jeXurePbvLf0aSSL18HGrOb1vWiCivmaHiGS4DmzPfL4rqmLD515D5XaiCLS5G0y8Oai8fufGqDiKfCSHCiXq0OfaL0CX9qy5ivXKajHvCz4mP1aSrH; logged_in=true; __ar_v4=2TZBGUFB4NG3XOXIT65JNC%3A20160905%3A2%7CZG5Z2TEJUZBPBFFAJOVBWT%3A20160905%3A2%7CBZOLVVOXJVEOFDOIOAFMVC%3A20160905%3A2; _mkto_trk=id:519-AYY-549&token:_mch-codenvy.com-1473077540921-59399; __utma=216772740.161833912.1473077541.1473082731.1473084710.2; __utmc=216772740; __utmz=216772740.1473082731.1.1.utmcsr=(direct)|utmccn=(direct)|utmcmd=(none); JSESSIONID=41F787DE369FC019C2D21CCE25158BC5; _gat_UA-37306001-3=1; _gat=1; _ga=GA1.3.161833912.1473077541; _ga=GA1.2.161833912.1473077541' -H 'Connection: keep-alive' --data-binary $'------WebKitFormBoundaryYpGppVWGMw0UYAj3\r\nContent-Disposition: form-data; name="factory"\r\n\r\n{"v":"4.0","workspace":{"defaultEnv":"default","environments":[{"name":"default","machineConfigs":[{"limits":{"ram":1000},"name":"default","type":"docker","source":{"type":"dockerfile","content":"FROM codenvy/ubuntu_jdk8"},"dev":true,"servers":[],"envVariables":{},"links":[]}]}],"projects":[{"links":[],"name":"codenvass","attributes":{},"type":"blank","source":{"location":"https://subversion.assembla.com/svn/codenvass/","type":"subversion","parameters":{"password":"s1o2m3e4","username":"Codenvass"}},"path":"/codenvass","problems":[],"mixins":[]}],"name":"wksp-v06o","commands":[{"commandLine":"mvn clean install -f ${current.project.path}","name":"build","attributes":{},"type":"mvn"},{"commandLine":"mvn -f /projects/web-java-spring clean install && cp /projects/web-java-spring/target/*.war $TOMCAT_HOME/webapps/ROOT.war","name":"web-java-spring: build","attributes":{"previewUrl":""},"type":"mvn"},{"commandLine":"$TOMCAT_HOME/bin/catalina.sh run","name":"web-java-spring: run tomcat","attributes":{"previewUrl":"http://${server.port.8080}"},"type":"custom"},{"commandLine":"$TOMCAT_HOME/bin/catalina.sh stop","name":"web-java-spring: stop tomcat","attributes":{"previewUrl":""},"type":"custom"},{"commandLine":"mvn -f /projects/web-java-spring clean install && cp /projects/web-java-spring/target/*.war $TOMCAT_HOME/webapps/ROOT.war && $TOMCAT_HOME/bin/catalina.sh run","name":"web-java-spring: build and run","attributes":{"previewUrl":"http://${server.port.8080}"},"type":"mvn"},{"commandLine":"mvn -f /projects/web-java-spring clean install && cp /projects/web-java-spring/target/*.war $TOMCAT_HOME/webapps/ROOT.war && $TOMCAT_HOME/bin/catalina.sh jpda run","name":"web-java-spring: debug","attributes":{"previewUrl":"http://${server.port.8080}"},"type":"mvn"}],"links":[]},"links":[]}\r\n------WebKitFormBoundaryYpGppVWGMw0UYAj3--\r\n' --compressed

**Comment by bmicklea** [comment]
Additional information on the customer's environment:
- All SVN repos are private
- All are secured with username/password

**Comment by bmicklea** [comment]
@vkuznyetsov @sleshchenko is there additional information you need to assess this?

**Comment by soanni** [comment]
we will really appreciate a quick fix, for us it's a huge stopper. Half of our products are in SVN :(

**Comment by bmicklea** [comment]
@vkuznyetsov this should be marked `sprint/current-sprint` now, right?

**Comment by bmicklea** [comment]
@vinokurig what's the latest status?

**Comment by vinokurig** [comment]
@bmicklea I am going to send a pull request for this tomorrow

**Comment by bmicklea** [comment]
Fantastic news - thanks Igor!

**Comment by bmicklea** [comment]
@vinokurig is the pull request completed?

**Comment by vinokurig** [comment]
@bmicklea yes, it is on review, changed the status

**Comment by bmicklea** [comment]
Thanks - can it be merged into master today or tomorrow?

**Comment by vinokurig** [comment]
I hope to it , it depends on review results

**Comment by bmicklea** [comment]
@eivantsov, Igor needs some changes to the docker images to make sure this works as expected. These are needed before we can release M5.

**Comment by bmicklea** [comment]
@vinokurig are the docker changes you needed complete as well?

**Comment by vinokurig** [comment]
This question is to @eivantsov, but I have tried this on new base image and it worked

**Comment by bmicklea** [comment]
Wonderful, thanks Igor!
